perl/ANNOTATION.md

```diff
@@ -0,0 +1,781 @@
+
+Multiallelics: Bystro decomposes multiallelic sites into separate rows, each with a single alternate allele. This ensures that each row has a single alternate allele, and that the annotations are accurate for each allele. The appropriate transformations are done to ensure that VCF padding rules are correctly transformed into Bystro's representation, and ensures that all variants outtputted in Bystro are left-aligned.
+
+<br/>
+
+### Annotation Fields for Default Human Assembly hg38 and hg19 Bystro Databases
+
+#### Basic Fields
+
+Sourced from the input file, or calculated based on input fields from the VCF or SNP file pre-processor.
+
+`chrom` - chromosome, always prepended with "chr"
+
+`pos` - genomic position after Bystro normalizes variant representations
+
+- Positions always correspond to the first affected base.
+
+`type` - the type of variant
+
+- VCF format types: `SNP`, `INS`, `DEL`, `MULTIALLELIC`
+  - Multi-nucleotide polymorphisms are decomposed into separate rows of `type` "SNP", each of 1 variant. In the next release of Bystro these will be called "MNP" and have a "linkage" property to enable you to reconstruct the original MNP, while still retaining the full set of per-SNP annotations
+  - Multiallelics are decomposed into separate rows, but retain the "MULTIALLELIC" `type`
+- SNP format types: `SNP`, `INS`, `DEL`, `MULTIALLELIC`, `DENOVO`
+
+`inputRef` - the reference base, as it is present in the input file, after variant and reference normalization.
+
+- This is generated by the input file pre-processor (hence `input` in the name), and is always 1 base long - the affected reference base at that position
+
+`alt` - the alternate/nonreference allele
+
+- VCF multi-allelic and MNP sites are decomposed into individual entries of a single allele.
+  - Genotypes are properly segregated per allele
+
+`trTv` - transition:transversion ratio for your dataset at this position
+
+- Possible values: `0`, `1`, `2`
+  - 0 indicates neither transition nor transversion (which occurs when the alternate allele is an insertion or deleetion)
+  - 1 is a transition (purine -> purine or pyrimidine -> pyrimidine)
+  - 2 is a transversion (pyridine -> pyrimidine or vice versa)
+
+`heterozygotes` - The array of heterozygous sample labels
+
+`heterozygosity` - The fraction of all samples (excluding missing samples) that are heterozygous for the alternate allele
+
+`homozygotes` - The array of homozygous sample labels
+
+`homozygosity` - The fraction of all samples that are homozygous for the alternate allele
+
+`missingGenos` - The samples that did not have a genotype (e.g., ".") at the site. If an individual has at least 1 missing genotype, they are considered missing for the site.
+
+- e.g., .|., .|0 are all considered missing
```

Comment:
I would drop the e.g., here unless you mean ',' is considered missing. I like the 'Possible values:' at line 91, I think that format is just a little clearer to grasp right away. Might be nice to have 'possible values' wherever it's just a few options like that, or 'example values' when there's more possibilities and it's easy to show an example